cardano-db-sync/src/Cardano/DbSync/DbAction.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+module Cardano.DbSync.DbAction
+  ( DbAction (..)
+  , DbActionQueue (..)
+  , MkDbAction (..)
+  , blockingFlushDbActionQueue
+  , lengthDbActionQueue
+  , newDbActionQueue
+  , writeDbActionQueue
+  ) where
+
+import           Cardano.Prelude
+
+import           Cardano.DbSync.Types
+
+import qualified Control.Concurrent.STM as STM
+import           Control.Concurrent.STM.TBQueue (TBQueue)
+import qualified Control.Concurrent.STM.TBQueue as TBQ
+
+import           Ouroboros.Consensus.Byron.Ledger (ByronBlock (..))
+import           Ouroboros.Network.Block (Point (..), Tip)
+
+
+data DbAction
+  = DbApplyBlock !CardanoBlockTip
+  | DbRollBackToPoint !CardanoPoint
+  | DbFinish
+
+
+newtype DbActionQueue = DbActionQueue
+  { dbActQueue :: TBQueue DbAction
+  }
+
+class MkDbAction blk where
```

Comment:
Nice, but I think the consensus guys are doing something pretty similar with the abstractions.

Reply:
Sorry, I do not think that is nice.